### PR TITLE
fix(Scripts/BlackTemple): Mother Shahraz Periodic part 2

### PR DESCRIPTION
**Pull request by Gultask**
## Changes Proposed:

This PR proposes changes to:
-  [ ] Core (units, players, creatures, game systems).
-  [ ] Scripts (bosses, spell scripts, creature scripts).
-  [ ] Database (SAI, creatures, etc).

## Issues Addressed:

- Closes https://github.com/chromiecraft/chromiecraft/issues/7241

## SOURCE:

The changes have been validated through:
- [ ] Live research (checked on live servers, e.g Classic WotLK, Retail, etc.)
- [ ] Sniffs (remember to share them with the open source community!)
- [ ] Video evidence, knowledge databases or other public sources (e.g forums, Wowhead, etc.)
- [ ] The changes promoted by this pull request come partially or entirely from another project (cherry-pick). **Cherry-picks must be committed using the proper --author tag in order to be accepted, thus crediting the original authors, unless otherwise unable to be found**

## Tests Performed:

This PR has been:
- [ ] Tested in-game by the author.
- [ ] Tested in-game by other community members/someone else other than the author/has been live on production servers.
- [ ] This pull request requires further testing and may have edge cases to be tested.

## How to Test the Changes:

- [ ] This pull request can be tested by following the reproduction steps provided in the linked issue
- [ ] This pull request requires further testing. Provide steps to test your changes. If it requires any specific setup e.g multiple players please specify it as well.

1.
2.
3.

## Known Issues and TODO List:

- [ ]
- [ ]

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Review by Nyeriah — APPROVED** [review]

File: src/server/scripts/Outland/BlackTemple/boss_mother_shahraz.cpp [hunk 1]
```diff
@@ -59,6 +59,7 @@ enum Spells
 
 enum Misc
 {
+    DATA_SHAHRAZ_BEAM               = 1,
```
Comment (Nyeriah) [inline src/server/scripts/Outland/BlackTemple/boss_mother_shahraz.cpp:62]: leftover?

**Comment by Gultask** [comment]
> The beam should change after the 3rd cast and not the 6th

yup, but I was mistaken when making that first PR
What is happening is that there's 3 spells here

Random Periodic -> Beam Periodic -> Beam Damage Spell

Random Periodic ticks every 6s. When it goes through, whether in the first tick or the 6th, it will apply a random 'Beam Periodic' on her, the latter will then tick every 9s, casting a Beam Damage Spell

File: src/server/scripts/Outland/BlackTemple/boss_mother_shahraz.cpp (hunk 1 above)
Comment (Gultask) [reply src/server/scripts/Outland/BlackTemple/boss_mother_shahraz.cpp]: ya

**Comment by Gultask** [comment]
I've thought about how it might work. It could be that it checks for if any of the four periodic auras are applied, if yes, it does nothing. Else, it applies a random periodic

**Comment by amed80** [comment]
atm it seems that it does use a beam 3 times then it always wait 18s before applying a new tick instead of 12-18s

**Comment by Gultask** [comment]
I'll merge this one because it's higher priority to undo the mess I made

**Pull request merged**